model/src/main/proto/survey.proto

```diff
@@ -0,0 +1,204 @@
+  string survey_id = 1;
+
+  // A list of questions that make up the survey.
+  repeated SurveyQuestion questions = 2;
+}
+
+// Represents a question that is part of the Survey.
+message SurveyQuestion {
+  // The ID of the survey question. This is a string value that uniquely identifies the question
+  // within the context of the survey, and is guaranteed to be unique within the survey scope only.
+  string question_id = 1;
+
+  // The semantic name of the question used to identify it in the analytics dashboard.
+  QuestionName question_name = 2;
+
+  // The actual text value of the question.
+  string question = 3;
+
+  // The type of question.
+  oneof question_type {
+    // Corresponds to a question with multiple choices.
+    SurveyQuestionOptionList multiple_choice = 4;
+
+    // Corresponds to a question with multiple choices and a free-text 'other' option.
+    SurveyQuestionOptionList multiple_choice_with_other = 5;
+
+    // Indicates whether this question takes a free-text response.
+    bool free_form_text = 6;
+  }
+
+  // The language code to which this question is localized.
+  string language_code = 7;
+}
+
+// Represents a list of answer options for a multiple choice question.
+message SurveyQuestionOptionList {
+  repeated SurveyQuestionOption options = 1;
+}
+
+// Corresponds to the semantic name given to each question.
+enum QuestionName {
+  // The question name is unknown.
+  QUESTION_NAME_UNSPECIFIED = 0;
+
+  // Corresponds to the name of the user type question.
+  USER_TYPE = 1;
+
+  //  Corresponds to the market fit question.
+  MARKET_FIT = 2;
+
+  // Corresponds to the NPS question.
+  NPS = 3;
+
+  // Corresponds to the open-ended question for a NPS promoter.
+  PROMOTER_FEEDBACK = 4;
+
+  // Corresponds to the open-ended question for a NPS passive.
+  PASSIVE_FEEDBACK = 5;
+
+  // Corresponds to the open-ended question for a NPS detractor.
+  DETRACTOR_FEEDBACK = 6;
+}
+
+// Represents the multiple choice answer available for a question.
+message SurveyQuestionOption {
+  // The ID of this question option. It is a string that uniquely identifies the question option
+  // within the context of a survey question. It is only unique within the scope of a survey.
+  string question_option_id = 1;
+
+  // The semantic name of the question this answer option is tied to.
+  QuestionName question_name = 2;
+
+  // The actual text value of the answer option.
+  string label = 3;
+
+  // The type of answer this option represents.
+  oneof answer {
+    // The enum representation of the type of user answer option.
+    UserTypeAnswer user_type = 4;
+
+    // The enum representation of the market fit answer option.
+    MarketFitAnswer market_fit = 5;
+
+    // The integer value representing the nps score option.
+    int32 nps = 6;
+  }
+}
+
+// Represents an answer that was selected by a user for a survey question.
+message SurveyUserAnswer {
```

Comment:
Should this maybe be ``SurveyQuestionAnswer``? The 'User' part is throwing me off--I think this is meant to have parity with ``SurveyQuestionOption`` above, so a rename might help connect them better.

Reply:
I'm inclined to go with `SurveySelectedAnswer` so that it can make sense also in the context of `EphemeralSurveyQuestion`

Reply:
That sounds good--thanks!